dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/JavaArchive.java

```diff
@@ -0,0 +1,575 @@
+				if (block.length < 4) {
+					throw new IOException("Reached end of file while searching for to Zip64-End-of-Central-Directory-Record does not correct.");
+				}
+				offsetInBlock = 0;
+			}
+		}
+
+		if (block.length < (offsetInBlock + ZIP64_EOCDR_MINIMUM_SIZE ))  {
+			block = getBlockFromBeginningOfFile(file, offsetFromBeginningOfFile, ZIP64_EOCDR_MINIMUM_SIZE);
+		    return  Arrays.copyOfRange(block, 0, ZIP64_EOCDR_MINIMUM_SIZE);
+		} 
+		
+		// Only store the minimum size for this record.  We don't need to store any extra data.
+		return Arrays.copyOfRange(block, offsetInBlock , offsetInBlock + ZIP64_EOCDR_MINIMUM_SIZE);
+	}
+	
+	/**
+	 * @param file a zip file
+	 * @param size number of bytes to read from end of file.
+	 * @return byte[] an array of data of 'size' bytes from the end of the file
+	 * @throws IOException
+	 */
+	private byte[] getBlockFromEndOfFile(RandomAccessFile file, int size) throws IOException {
+
+		int blockLength =  size >_fileLength ? (int)_fileLength : size;
+		long offset = _fileLength - blockLength;
+
+		file.seek(offset);
+
+		byte[] bytes = new byte[blockLength];
+		file.read(bytes, 0, blockLength );
+		return bytes;
+	}
+	
+	/**
+	 * 
+	 * @param file  a zip file
+	 * @param offsetFromEnd Offset from end of file where reading begins
+	 * @param size number of bytes of data to read
+	 * @return Returns an array of 'size' bytes that starts from 'offsetFromEnd' of file.
+	 * @throws IOException
+	 */
+	private byte[] getBlockFromEndOfFile(RandomAccessFile file, int offsetFromEnd, int size) throws IOException {
+		
+		// Have offset from end.  Get offset from beginning.
+		long offset = _fileLength >= offsetFromEnd ? _fileLength - offsetFromEnd : 0;
+
+		file.seek(offset);
+
+		byte[] bytes = new byte[size];
+		file.read(bytes, 0, size );
+		return bytes;
+	}
+
+	/**
+	 * 
+	 * @param file a zip file
+	 * @param offset Position from beginning of file, where data is to be read
+	 * @param size number of bytes to read
+	 * @return Returns an array of size 'size' read from file beginning at 'offset'
+	 * @throws IOException
+	 */
+	private byte[] getBlockFromBeginningOfFile(RandomAccessFile file, long offset, int size) throws IOException {
+		
+		file.seek(offset);
+
+		byte[] bytes = new byte[size];
+		file.read(bytes, 0, size );
+		return bytes;	
+	}
+	
+	/** 
+	 * Test for valid signature and size.
+	 * @return true if the offset points to a valid End-of-Central-Directory-Record within "bytes"
+	 */
+	private boolean isValidEoCDRec(byte[] bytes, int offset, int centralDirEndRecSize) {
+		if ( !isSignature(bytes, offset, EOCDR_SIGNATURE) 
+			 || !isValidEoCDRecSize(bytes, offset + EOCDR_COMMENT_LENGTH_OFFSET, centralDirEndRecSize)) {
+			return false;
+		}
+        return true;
+	}
+	
+	/**
+	 * 
+	 * @return true if the signature parameter matches the bytes at offset
+	 */
+	private boolean isSignature(byte[] bytes, int offset, int signature) {
+
+		long value = getSignature(bytes, offset);
+
```

Comment:
We are reading a long but comparing that against an int signature value?

Reply:
Changed getSignature to return an int.